sympy/polys/matrices/ddm.py

```diff
@@ -598,22 +598,55 @@ def rref_den(a):
         return b, denom, pivots
 
     def nullspace(a):
+        """Returns a basis for the nullspace of a.
+
+        The domain of the matrix must be a field.
+
+        See Also
+        ========
+
+        rref
+        sympy.polys.matrices.domainmatrix.DomainMatrix.nullspace
+        """
         rref, pivots = a.rref()
-        rows, cols = a.shape
-        domain = a.domain
+        return rref.nullspace_from_rref(pivots)
+
+    def nullspace_from_rref(a, pivots):
+        """Compute the nullspace of a matrix from its rref.
+
+        The domain of the matrix can be any domain.
+
+        Returns a tuple (basis, nonpivots).
+
+        See Also
+        ========
+
+        sympy.polys.matrices.domainmatrix.DomainMatrix.nullspace
+            The higher level interface to this function.
+        """
+        m, n = a.shape
+        K = a.domain
+
+        if not pivots:
+            return (a.eye(n, K), list(range(n)))
+
+        # After rref the pivots are all one but after rref_den they may not be.
+        pivot_val = a[0][pivots[0]]
```

Comment:
Does `pivot_val` stay some into some constant value for `rref_den`?

Reply:
All pivots will always have the same value and will be equal to the denominator:
```python
In [12]: M = randMatrix(4)

In [13]: M.to_DM().rref_den()[0].to_Matrix()
Out[13]: 
⎡-13677854      0          0          0    ⎤
⎢                                          ⎥
⎢    0      -13677854      0          0    ⎥
⎢                                          ⎥
⎢    0          0      -13677854      0    ⎥
⎢                                          ⎥
⎣    0          0          0      -13677854⎦
```
If the matrix does not fully reduce to the identity then it is usually not possible to factor out the pivot value because of the non-pivot entries:
```python
In [18]: M = Matrix([[1, 2, 3], [4, 5, 7]])

In [19]: M.to_DM().rref_den()[0].to_Matrix()
Out[19]: 
⎡-3  0   1 ⎤
⎢          ⎥
⎣0   -3  -5⎦
```
Here the pivots have value `-3` but that does not divide 5 so we cannot factor `-3` out  of the matrix.